### PR TITLE
feat(completion): add getCompletionEntryDetails and getCompletionsAtPosition support

### PR DESCRIPTION
**Pull request by pikax**
Missing tests

### CODE REVIEW

**Review by znck — APPROVED** [review]

**Comment by znck** [comment]
Can you run `pnpm install` to update lockfile?

**Comment by pikax** [comment]
I'm still doing the tests, having some issues 😄

File: packages/typescript-plugin-vue/src/servers/routing.ts [hunk 1]
```diff
@@ -260,6 +260,31 @@ function createLanguageServiceRouter(
           })
           .filter(isNotNull)
       },
+      getCompletionsAtPosition(fileName, position, options) {
+        return choose(fileName).getCompletionsAtPosition(
```
Comment (znck) [inline packages/typescript-plugin-vue/src/servers/routing.ts:264]: We may need to remove references to virtual files from completions.

Comment (pikax) [reply packages/typescript-plugin-vue/src/servers/routing.ts:264]: how can I filter those?

Example of entries from the test:

```json
[
  {
    "name": "compile",
    "kind": "function",
    "kindModifiers": "export,declare",
    "sortText": "0"
  },
  {
    "name": "useCssModule",
    "kind": "function",
    "kindModifiers": "export,declare",
    "sortText": "0"
  },
  {
    "name": "useCssVars",
    "kind": "function",
    "kindModifiers": "export,declare",
    "sortText": "0"
  },
  {
    "name": "createApp",
    "kind": "const",
    "kindModifiers": "export,declare",
    "sortText": "0"
  },
  {
    "name": "createSSRApp",
    "kind": "const",
    "kindModifiers": "export,declare",
    "sortText": "0"
  },
  {
    "name": "hydrate",
    "kind": "const",
    "kindModifiers": "export,declare",
    "sortText": "0"
  },
  {
    "name": "render",
    "kind": "const",
    "kindModifiers": "export,declare",
    "sortText": "0"
  },
  {
    "name": "Transition",
    "kind": "const",
    "kindModifiers": "export,declare",
    "sortText": "0"
  },
  {
    "name": "TransitionGroup",
    "kind": "const",
    "kindModifiers": "export,declare",
    "sortText": "0"
  },
  {
    "name": "TransitionGroupProps",
    "kind": "type",
    "kindModifiers": "export,declare",
    "sortText": "0"
  }
]
```

Comment (znck) [reply packages/typescript-plugin-vue/src/servers/routing.ts:264]: I don't think we need to filter these, we have to correct fileName for import suggestions from .vue (if any)

File: test/baseline/completioninfo.spec.ts [hunk 2]
```diff
@@ -0,0 +1,207 @@
...
+    })
+
+    it('should provide info on import vue', async () => {
+      const { body } = await server.sendCommand(
+        'completionInfo',
+        await findPositionOrThrowIn(file, `import {} from 'vue'`, 8),
+      )
+
+      expect(body?.entries.length).toBeGreaterThan(1)
+      expect(body?.entries).toContainEqual({
```
Comment (pikax) [inline test/baseline/completioninfo.spec.ts:54]: this actually returns +200 entries! 

I reckon the typescript language server will then filter them, but not sure

Comment (znck) [reply test/baseline/completioninfo.spec.ts:54]: We can look into this later, in current state it's a great addition to feature set.

Comment (pikax) [reply test/baseline/completioninfo.spec.ts:54]: It works on the actual editor, on the logs it also runs the command: `getCompletionEntryDetails`

**Comment by pikax** [comment]
something is off with the CI 😕

File: packages/typescript-plugin-vue/src/servers/routing.ts (hunk 1 above)
Comment (pikax) [reply packages/typescript-plugin-vue/src/servers/routing.ts]: added that 😎 
https://github.com/znck/vue-developer-experience/pull/31/commits/606cd0c4c586691d50aedad581a3ecf0fbe8c437

**Comment by znck** [comment]
Yeah looks like codecov is failing, I think we should lock it's version.

**Review by znck — APPROVED** [review]

**Pull request merged**